### ISSUE TITLE
[XMaterial] - Unsupported material from item: MAP

### ISSUE THREAD

**Issue by xGinko** [CryptoMorin/XSeries#203]
**Description**
Pastebin: https://pastebin.com/bHw92rjM

Not entirely sure why this is happening, the method involved in this is as follows:

```
 public static boolean isSpawnEgg(ItemStack item) {
        if (item == null) return false;
        switch (XMaterial.matchXMaterial(item)) {
            case ALLAY_SPAWN_EGG:
            case AXOLOTL_SPAWN_EGG:
            case BAT_SPAWN_EGG:
            case BEE_SPAWN_EGG:
            case BLAZE_SPAWN_EGG:
            case CAMEL_SPAWN_EGG:
            case CAT_SPAWN_EGG:
            case CAVE_SPIDER_SPAWN_EGG:
            case CHICKEN_SPAWN_EGG:
            case COD_SPAWN_EGG:
            case COW_SPAWN_EGG:
            case CREEPER_SPAWN_EGG:
            case DOLPHIN_SPAWN_EGG:
            case DONKEY_SPAWN_EGG:
            case DROWNED_SPAWN_EGG:
            case ELDER_GUARDIAN_SPAWN_EGG:
            case ENDERMAN_SPAWN_EGG:
            case ENDERMITE_SPAWN_EGG:
            case ENDER_DRAGON_SPAWN_EGG:
            case EVOKER_SPAWN_EGG:
            case FOX_SPAWN_EGG:
            case FROG_SPAWN_EGG:
            case GHAST_SPAWN_EGG:
            case GLOW_SQUID_SPAWN_EGG:
            case GOAT_SPAWN_EGG:
            case GUARDIAN_SPAWN_EGG:
            case HOGLIN_SPAWN_EGG:
            case HORSE_SPAWN_EGG:
            case HUSK_SPAWN_EGG:
            case IRON_GOLEM_SPAWN_EGG:
            case LLAMA_SPAWN_EGG:
            case MAGMA_CUBE_SPAWN_EGG:
            case MOOSHROOM_SPAWN_EGG:
            case MULE_SPAWN_EGG:
            case OCELOT_SPAWN_EGG:
            case PANDA_SPAWN_EGG:
            case PARROT_SPAWN_EGG:
            case PHANTOM_SPAWN_EGG:
            case PIGLIN_BRUTE_SPAWN_EGG:
            case PIGLIN_SPAWN_EGG:
            case PIG_SPAWN_EGG:
            case PILLAGER_SPAWN_EGG:
            case POLAR_BEAR_SPAWN_EGG:
            case PUFFERFISH_SPAWN_EGG:
            case RABBIT_SPAWN_EGG:
            case RAVAGER_SPAWN_EGG:
            case SALMON_SPAWN_EGG:
            case SHEEP_SPAWN_EGG:
            case SHULKER_SPAWN_EGG:
            case SILVERFISH_SPAWN_EGG:
            case SKELETON_HORSE_SPAWN_EGG:
            case SKELETON_SPAWN_EGG:
            case SLIME_SPAWN_EGG:
            case SNOW_GOLEM_SPAWN_EGG:
            case SPIDER_SPAWN_EGG:
            case SQUID_SPAWN_EGG:
            case STRAY_SPAWN_EGG:
            case STRIDER_SPAWN_EGG:
            case TADPOLE_SPAWN_EGG:
            case TRADER_LLAMA_SPAWN_EGG:
            case TROPICAL_FISH_SPAWN_EGG:
            case TURTLE_SPAWN_EGG:
            case VEX_SPAWN_EGG:
            case VILLAGER_SPAWN_EGG:
            case VINDICATOR_SPAWN_EGG:
            case WANDERING_TRADER_SPAWN_EGG:
            case WARDEN_SPAWN_EGG:
            case WITCH_SPAWN_EGG:
            case WITHER_SKELETON_SPAWN_EGG:
            case WITHER_SPAWN_EGG:
            case WOLF_SPAWN_EGG:
            case ZOGLIN_SPAWN_EGG:
            case ZOMBIE_HORSE_SPAWN_EGG:
            case ZOMBIE_SPAWN_EGG:
            case ZOMBIE_VILLAGER_SPAWN_EGG:
            case ZOMBIFIED_PIGLIN_SPAWN_EGG:
                return true;
            default:
                return false;
        }
    }
```

All I understand here that it does not know "MAP" as a material name? Please advice. 
Could this also be due something like the player replacing the plugin in the plugin folder without restarting the server?

**Version**
Server version: Paper 1.12.2, XSeries version 9.2.0.
Maven import looks like this: 

```
<dependency>
    <groupId>com.github.cryptomorin</groupId>
    <artifactId>XSeries</artifactId>
    <version>9.2.0</version>
</dependency>
```

api in plugin.yml is set to 1.13,

project code can currently be found here:
https://github.com/xGinko/AnarchyExploitFixes/

**Comment by CryptoMorin** [comment]
The data value for your map (i.e. legacy `ItemStack#getDamage()`) is returning a negative number. That's not supposed to happen. I need to know how to reproduce it.

**Comment by LOOHP** [comment]
From what I know there are map art plugins that utilise "negative map ids" to "not affect your server's vanilla maps" as a selling point. It might be an instance of that.

**Comment by xGinko** [comment]
Could this be circumvented by using matchXmaterial(item.getType())?

**Comment by CryptoMorin** [comment]
> Could this be circumvented by using matchXmaterial(item.getType())?
> 
> Also I'm sorry I do not know how to reproduce it as this error has been sent to me by a third party. In the past I thought it could have something to do with ItemStack.setData() in 1.12 but I failed trying to get an item with an illegal data value.

It can fix it, but using that method isn't recommended. Anyways, I think LOOHP is right, the map materials were special to begin with, I'll just make another exception for this case too.

**Comment by CryptoMorin** [comment]
Fixed in v9.3.0
